src/modals.ts

```diff
@@ -90,4 +92,37 @@ class CreateFolder extends Modal {
                     })
             })
     }
-}
\ No newline at end of file
+}
+
+export class VaultSelectionModal extends FuzzySuggestModal<string> {
+    rootDirectory: string;
+    app: App;
+    plugin: VaultTransferPlugin; // Ensure this plugin reference is passed in the constructor
+
+    constructor(app: App, plugin: VaultTransferPlugin, rootDirectory: string) {
+        super(app);
+        this.rootDirectory = rootDirectory;
+        this.plugin = plugin;
+    }
+
+    getItems(): string[] {
+        return listPossibleVaults(this.rootDirectory);
+    }
+
+    getItemText(item: string): string {
+        return item;
+    }
+
+    onChooseItem(item: string, evt: MouseEvent | KeyboardEvent): void {
+        console.log('Selected Vault:', item);
+        this.plugin.settings.outputVault = path.join(this.rootDirectory, item); // Adjust the outputVault setting
+        const editor = this.app.workspace.getActiveViewOfType(MarkdownView)?.editor;
+        const file = this.app.workspace.getActiveFile();
+        if (editor && file) {
+            transferNote(editor, file, this.app, this.plugin.settings);
+        } else {
+            showNotice("No active editor or file found.");
+        }
```

Comment:
Rather than update the shared instance of the user's settings, could we clone them for this transfer?
```suggestion
        const tempSettings = { ...this.plugin.settings };
        tempSettings.outputVault = path.join(this.rootDirectory, item);
        const editor = this.app.workspace.getActiveViewOfType(MarkdownView)?.editor;
        const file = this.app.workspace.getActiveFile();
        if (editor && file) {
            transferNote(editor, file, this.app, tempSettings);
        } else {
            showNotice("No active editor or file found.");
        }
```